nodes/monitor.go

```diff
@@ -381,10 +407,21 @@ func cleanHashes(hashdir string, skip []common.Hash) {
 //  Search uses binary search to find and return the smallest index i
 //  in [0, n) at which f(i) is true
 func findSplit(num int, a Node, b Node) int {
-	splitBlock := sort.Search(num, func(i int) bool {
-		return a.HashAt(uint64(i), false) != b.HashAt(uint64(i), false)
+	for i := len(headListCache) - 1; i > 0; i-- {
+		head := headListCache[i]
+		if a.HashAt(uint64(head), false) != b.HashAt(uint64(head), false) {
```

Comment:
```suggestion
		if aBlock, bBlock := a.HashAt(uint64(head), false), b.HashAt(uint64(head), false); aHead != bHead{
			// difference at 'head'
			if aParent, bParent := a.HashAt(uint64(head-1), false), b.HashAt(uint64(head-1), false); aParent == bParent{
				// parent is identical
				return head // forked at this block
			}
                 }
```
This should fix any the cornercase I mentioned (need to test with forks at '0' though, e.g. ropsten-node versus mainnet node, to check that we don't go out of limits